cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_check_manager.py

```diff
@@ -0,0 +1,375 @@
+    # The command in this subprocess call is built as literal
+    result = subprocess.run(
+        shlex.split(f"cat {str(file_path)}"),
+        timeout=FILE_READ_TIMEOUT,
+        stdout=subprocess.PIPE,
+        encoding="utf-8",
+        check=False,
+        shell=False,  # nosec B603
+    )
+    return result.stdout if hasattr(result, "stdout") else ""
+
+
+class HealthCheckManagerConfig:
+    """Class that identifies the Health Check Manager configuration."""
+
+    DEFAULTS = {
+        "health_check_timeout": 600,
+        "logging_config": os.path.join(os.path.dirname(__file__), "logging", "health_check_manager_logging.conf"),
+        "managed_health_check_dir": os.path.join(os.path.dirname(__file__), "health_checks"),
+    }
+
+    def __init__(self, config_file_path):
+        self._get_config(config_file_path)
+
+    def __repr__(self):
+        """Return Health Check Manager configuration representation."""
+        attrs = ", ".join([f"{key}={repr(value)}" for key, value in self.__dict__.items()])
+        return f"{self.__class__.__name__}({attrs})"
+
+    def __eq__(self, other):
+        """Return true if both objects are equal."""
+        if type(other) is type(self):
+            return self._config == other._config
+        return False
+
+    def __ne__(self, other):
+        """Return true if both objects are not equal."""
+        return not self.__eq__(other)
+
+    def _get_config(self, config_file_path: str):
+        """Get health check manager configuration."""
+        log.info("Reading '%s'", config_file_path)
+
+        self._config = ConfigParser()
+        try:
+            self._config.read_file(StringIO(_read_file(config_file_path)))
+        except Exception as err:
+            if hasattr(err, "message"):
+                err = err.message
+            log.error(
+                "Cannot read Health Check Manager config file: %s, falling back to default settings. Error is: %s",
+                config_file_path,
+                err,
+            )
+
+        self.health_check_timeout = self._config.getint(
+            "health_check_manager", "health_check_timeout", fallback=self.DEFAULTS.get("health_check_timeout")
+        )
+        self.logging_config = self._config.get(
+            "health_check_manager", "logging_config", fallback=self.DEFAULTS.get("logging_config")
+        )
+        self.managed_health_check_dir = self._config.get(
+            "health_check_manager", "managed_health_check_dir", fallback=self.DEFAULTS.get("managed_health_check_dir")
+        )
+
+        log.debug(repr(self))
+
+
+@dataclass
+class HealthCheckDefinition:
+    """
+    Health Check definition.
+
+    Contains all the configuration relevant to a specific health check.
+    """
+
+    name: str
+    is_managed: bool
+    is_enabled: bool
+    check_path: str
+    # custom_args: str
+    # action_on_failure: str
+    # execution_time: str
+    # url: str
```

Comment:
should you remove these commented fields?

Reply:
removed